x/oracle/types/genesis.go

```diff
@@ -10,22 +10,22 @@ import (
 // GenesisState - all clp state that must be provided at genesis
 //TODO: Add parameters to Genesis state ,such as minimum liquidity required to create a pool
 type GenesisState struct {
-	AddressWhitelist []sdk.ValAddress `json:"address_whitelist"`
-	AdminAddress     sdk.AccAddress   `json:"admin_address"`
+	AddressWhitelist map[string][]sdk.ValAddress `json:"address_whitelist"`
+	AdminAddress     sdk.AccAddress              `json:"admin_address"`
 }
 
 // NewGenesisState creates a new GenesisState instance
 func NewGenesisState() GenesisState {
 	return GenesisState{
-		AddressWhitelist: []sdk.ValAddress{},
+		AddressWhitelist: map[string][]sdk.ValAddress{},
```

Comment:
instead of mapping a network descriptor to an array of addresses, let's map a network descriptor to an address and map that address to a uint that is that users power.